### ISSUE TITLE
Proposal : Deployment ssh-keys support

### ISSUE THREAD

**Issue by iammichiel** [drone/drone#1891]
Hi there, 

I have been struggling to setup drone using private dependencies. I have found another issue mentioning a automatically generated `.netrc` to clone those dependencies over `git+https`. This is, in my opinion, not an acceptable solution. That would in my case, require updating a lot of other processes based on that repository. (deployment scripts, automated scaling process, setup scripts, documentation, etc)

A rather cleaner solution would be to implement deployment keys as most continuous integration systems handle them. 
- Travis presents different solutions here : https://docs.travis-ci.com/user/private-dependencies/
- CircleCI has a proper SSH Keys functionality as presented here : https://circleci.com/docs/configuration/#ssh-keys
- Bitbucket Pipelines does not implement it yet but suggests using variables to manually create the SSH keys as part of the process. 

In the case of Drone implementation, this would require two additional steps : 
- A configuration option to setup the deployment key (through web-ui, drone.yml, dronerc?)
- Injecting the deployment key in the drone running container 

What do you think of this?

**Comment by iammichiel** [comment]
I think I have not been clear enough about my usecase... 
For example, when using PHP, you generally manage dependency using Composer. And you can host private package/dependencies on a git repository. When setting up my build, I want to : 
- clone the initial repository (handled by drone)
- install all dependencies (handled by composer)

It is for the second step, I need to setup specific SSH keys to clone those repositories. 
I guess the problem is not only present for PHP packages.

**Comment by tboerger** [comment]
This still happens inside of plugins. The repository gets cloned by [drone-git](https://github.com/drone-plugins/drone-git), this plugin won't execute anything composer related, this will happen within some other docker container used in the pipeline. There you can inject secret variables however you want, including SSH keys.

**Comment by iammichiel** [comment]
Yes, I am very well aware of that. However, what I am suggesting is when starting a new container for the next build step, Drone automatically injects configured SSH keys. That functionality is described over here : https://circleci.com/docs/permissions-and-access-during-deployment/

Secrets are a work-around that might work somehow but makes the whole process quite complicated.

**Comment by bradrydzewski** [comment]
The drone secret store supports ssh keys
http://readme.drone.io/0.5/secrets/

See the accepted stackoverflow answer for an example:
http://stackoverflow.com/questions/41330947/how-to-create-a-drone-secret-file

> Secrets are a work-around that might work somehow but makes the whole process quite complicated.

I disagree. It adds a single line to your yaml file.

If you don't want to sign the yaml you can use the `--skip-verify` flag. If you want the secret available to multiple images you can use `--image=*`. If you only want to add the secret once and have it available to multiple repositories you can use organization or global secrets.

The bottom line is that Drone previously used deployment keys in 0.2 and this feature was explicitly removed and replaced with the netrc file. This has been, overall, a major improvement and is not something we are going to rollback.

**Comment by nlra** [comment]
Does this actually still work?

I've been trying for an entire day to just `npm install` a repo that has a private github repo, and it just appears impossible.

I tried adding the ssh key to the secrets as you quote above, I even output it to the logs in the pipeline to check that it works, so it's there. And yet, I still get `npm ERR! Permission denied (publickey).`. Which I assume means the ssh key isn't reallly used even after adding it to `/root/.ssh`?

PS: The netrc solution is nice, but in my situation I can only use the `git+ssh` syntax in my `package.json`, due to other business requirements.

Am I supposed to use a different directory or something?

Honestly this should be natively supported, it just doesn't make sense to have to use those weird hacks to use something as basic as an SSH key. Even if it's optional, there should be a way to automatically inject SSH keys into pipelines.
